### DISCUSSION TITLE
Printing float or integer to console

### DISCUSSION THREAD

**Discussion by Pandagineer** [bitcraze/discussions#320] [Q&A] [answered]
Hello,

In app_hello_world, let's say I wanted to print a float or integer (rather than string) to the Console. I tried this by inserting the lines:

float x;
x = 3.0;
DEBUG_PRINT(x);

But when making, I got an error. It seems that DEBUG_PRINT only accepts a string. My question is: in C#, how do I convert an integer or float to a string? (In Python, this is easy using the str() function.)

Thanks!

**Comment by ataffanel** [comment, marked as answer]
Hi!

First of all, the Crazyflie firmware is not written in C# but in C. The difference can be quite big between the two languages.

The ```DEBUG_PRINT()``` macro is emulating the standard C ```printf()``` fonction. The first argument is a [format string](https://en.wikipedia.org/wiki/Printf_format_string) and then a variable amount of arguments for the data. Your (expanded) example can be implemented by:

``` c
float x;
x = 3.0;
int y;
y = 42;
DEBUG_PRINT("%f %d", (double)x, y);   // This will print "3.0 42"
```

Note that not all the standard C format syntax will work, but the most common/simple will.

**Reply by Pandagineer** [reply]
My college C classes are all rushing back! 

This worked. Thank you. (I did not realize C# was a different language. I appreciate the correction.)
